### PR TITLE
✨ Change Image Uploads to S3 (#723)

### PR DESCRIPTION
**Pull request by mistryrn**
Bit of a WIP, still need to add a migration script for existing images stored in mongo, but figured this was a good way to park my progress before going on vacation.

Includes:
* add `aws-sdk` and `uuid` as dependencies to `cms`
* add env vars for: `S3_BUCKET`, `IAM_USER_KEY`, `IAM_USER_SECRET`
* upload images to S3, storing return URLs in new `file_url` in FilesSchema
* display images uploaded to S3 on the front-end
* updates to search query to fetch newly added `file_url`

### CODE REVIEW

File: cms/src/routes/images.js [hunk 1]
```diff
@@ -18,6 +23,26 @@ conn.once('open', function() {
   console.log('GridFSBucket is ready for uploads');
 });
 
+const s3 = new aws.S3({
```
Comment (joneubank) [inline cms/src/routes/images.js:26]: Can we move the S3 config to a module of its own in the /services directory? We can import it here for use.

File: cms/src/routes/images.js [hunk 2]
```diff
@@ -18,6 +23,26 @@ conn.once('open', function() {
...
+  accessKeyId: IAM_USER_KEY,
+  secretAccessKey: IAM_USER_SECRET,
+});
+
+const uploadToS3 = (fileStream, fileName) => {
```
Comment (joneubank) [inline cms/src/routes/images.js:31]: This is a reasonable method to export from the S3 service.

File: cms/src/routes/images.js [hunk 3]
```diff
@@ -71,25 +95,34 @@ imagesRouter.get('/:id', async (req, res) => {
...
-        code: 200,
-        msg: `image with id ${id} deleted`,
-      };
-    } catch (e) {
-      return {
-        code: 500,
-        msg: `error removing image with id ${id}`,
-      };
-    }
+    s3.deleteObject(params, (err, data) => {
```
Comment (joneubank) [inline cms/src/routes/images.js:103]: Create a `deleteFromS3(id)` method and export from S3 service

File: cms/src/helpers/migrateImages.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  },
+);
+
+conn.once('open', async () => {
+  try {
+    const images = await conn.db.collection('images.files').find({});
+    const bucket = await new mongoose.mongo.GridFSBucket(conn.db, { bucketName: 'images' });
+    console.log('Connection to db is ready, initiating image migration to S3...');
+
+    let migrations = [];
```
[the remark this answers is not in the document]
Comment (mistryrn) [reply cms/src/helpers/migrateImages.js:46]: Whoops 😅 Nice catch!

**Comment by mistryrn** [comment]
Updated PR to address comments from feedback :)

Key changes:
* S3-related code is now inside a reusable service
* image migration script is moved to /scripts

To run the migration script, make sure you add your IAM key and secret to the pm2 config, then from within /scripts you can run `ENV=prd node migrateImages.js`

Image uploads and deletes will also require the bucket/key/secret to be set up appropriately in your environment.

**Pull request merged**